src/pure-assert.js

```diff
@@ -55,40 +55,118 @@ export class UserError extends Error {
  * Throws a provided error if the second argument isn't trueish.
  * @param {Object} errorCls
  * @param {T} shouldBeTruthy
- * @param {string} message
+ * @param {string} opt_message
+ * @param {...*} var_args Arguments substituted into %s in the message
  * @return {T}
- * @throws {Error} when attribute values are missing or invalid.
+ * @throws {Error} when shouldBeTruthy is not truthy.
  */
-function pureAssertion(errorCls, shouldBeTruthy, message) {
+function pureAssertion(errorCls, shouldBeTruthy, opt_message, var_args) {
   // TODO: Support format strings.
-  if (!shouldBeTruthy) {
-    throw new errorCls(message);
+  if (shouldBeTruthy) {
+    return shouldBeTruthy;
   }
-  return shouldBeTruthy;
+
+  // Substitute provided values into format string in message
+  const message = Array.prototype.slice
```

Comment:
There is one thing this does not preserve: logic which detects if an element is passed and transforms the resulting error message to contain extra element info, adds `associatedElement` property, adds `fromAssert` property, etc. This felt AMP-centric and not necessarily needed for a standalone helper. Can add back in now or down the line if deemed necessary

Reply:
Compare to https://github.com/ampproject/amphtml/blob/master/src/log.js#L389

Reply:
That might actually have significant value for `useAssert`. But definitely has zero value for `devAssert`. Another reason why they are not all that similar.